### ISSUE TITLE
Dictionaries nullable segfault

### ISSUE THREAD

**Issue by filimonov** [ClickHouse/ClickHouse#8352]
Relates https://github.com/ClickHouse/ClickHouse/issues/4910

Dictionary with clickhouse as a source, some columns are declared as Nullable in ClickHouse underlying table (attributes, not keys), but without Nullable in dictionary xml configuration.

```
2019.12.22 19:54:38.941515 [ 189 ] {} <Fatal> BaseDaemon: ########################################
2019.12.22 19:54:38.942032 [ 189 ] {} <Fatal> BaseDaemon: (version 19.17.4.11 (official build)) (from thread 101) Received signal Segmentation fault (11).
2019.12.22 19:54:38.942093 [ 189 ] {} <Fatal> BaseDaemon: Address: 0x10010 Access: read. Address not mapped to object.
2019.12.22 19:54:38.942135 [ 189 ] {} <Fatal> BaseDaemon: Stack trace: 0x55986b0870d0 0x5598674f8969 0x5598674f93f6 0x5598674fab1e 0x5598674fb0bb 0x5598674fca01 0x559868c78e6a 0x559868e2e774 0x559868e2ebcd 0x559868e321bf 0x559868e35eef 0x559868e331d7 0x559865aff51c 0x55986b884e00 0x7f39feed86db 0x7f39fe7f588f
2019.12.22 19:54:38.984078 [ 189 ] {} <Fatal> BaseDaemon: 3. 0x55986b0870d0 memcpy /usr/bin/clickhouse
2019.12.22 19:54:38.984135 [ 189 ] {} <Fatal> BaseDaemon: 4. 0x5598674f8969 DB::ComplexKeyHashedDictionary::setAttributeValue(DB::ComplexKeyHashedDictionary::Attribute&, StringRef, DB::Field const&) /usr/bin/clickhouse
2019.12.22 19:54:38.984147 [ 189 ] {} <Fatal> BaseDaemon: 5. 0x5598674f93f6 DB::ComplexKeyHashedDictionary::blockToAttributes(DB::Block const&) /usr/bin/clickhouse
2019.12.22 19:54:38.984159 [ 189 ] {} <Fatal> BaseDaemon: 6. 0x5598674fab1e DB::ComplexKeyHashedDictionary::loadData() /usr/bin/clickhouse
2019.12.22 19:54:38.984185 [ 189 ] {} <Fatal> BaseDaemon: 7. 0x5598674fb0bb DB::ComplexKeyHashedDictionary::ComplexKeyHashedDictionary(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, DB::DictionaryStructure const&, std::unique_ptr<DB::IDictionarySource, std::default_delete<DB::IDictionarySource> >, DB::ExternalLoadableLifetime, bool, std::shared_ptr<DB::Block>) /usr/bin/clickhouse
2019.12.22 19:54:38.984198 [ 189 ] {} <Fatal> BaseDaemon: 8. 0x5598674fca01 ? /usr/bin/clickhouse
2019.12.22 19:54:38.984219 [ 189 ] {} <Fatal> BaseDaemon: 9. 0x559868c78e6a DB::DictionaryFactory::create(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Poco::Util::AbstractConfiguration const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, DB::Context const&) const /usr/bin/clickhouse
2019.12.22 19:54:38.984240 [ 189 ] {} <Fatal> BaseDaemon: 10. 0x559868e2e774 DB::ExternalDictionariesLoader::create(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Poco::Util::AbstractConfiguration const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const /usr/bin/clickhouse
2019.12.22 19:54:38.984277 [ 189 ] {} <Fatal> BaseDaemon: 11. 0x559868e2ebcd DB::ExternalLoader::createObject(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, DB::ExternalLoader::ObjectConfig const&, bool, std::shared_ptr<DB::IExternalLoadable const> const&) const /usr/bin/clickhouse
2019.12.22 19:54:38.984312 [ 189 ] {} <Fatal> BaseDaemon: 12. 0x559868e321bf std::_Function_handler<std::shared_ptr<DB::IExternalLoadable const> (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, DB::ExternalLoader::ObjectConfig const&, bool, std::shared_ptr<DB::IExternalLoadable const> const&), std::_Bind<std::shared_ptr<DB::IExternalLoadable const> (DB::ExternalLoader::*(DB::ExternalLoader*, std::_Placeholder<1>, std::_Placeholder<2>, std::_Placeholder<3>, std::_Placeholder<4>))(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, DB::ExternalLoader::ObjectConfig const&, bool, std::shared_ptr<DB::IExternalLoadable const> const&) const> >::_M_invoke(std::_Any_data const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, DB::ExternalLoader::ObjectConfig const&, bool&&, std::shared_ptr<DB::IExternalLoadable const> const&) /usr/bin/clickhouse
2019.12.22 19:54:38.984336 [ 189 ] {} <Fatal> BaseDaemon: 13. 0x559868e35eef DB::ExternalLoader::LoadingDispatcher::doLoading(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long, bool) /usr/bin/clickhouse
2019.12.22 19:54:38.984359 [ 189 ] {} <Fatal> BaseDaemon: 14. 0x559868e331d7 ThreadFromGlobalPool::ThreadFromGlobalPool<void (DB::ExternalLoader::LoadingDispatcher::*)(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long, bool), DB::ExternalLoader::LoadingDispatcher*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long&, bool>(void (DB::ExternalLoader::LoadingDispatcher::*&&)(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long, bool), DB::ExternalLoader::LoadingDispatcher*&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long&, bool&&)::{lambda()#1}::operator()() const /usr/bin/clickhouse
2019.12.22 19:54:38.984374 [ 189 ] {} <Fatal> BaseDaemon: 15. 0x559865aff51c ThreadPoolImpl<std::thread>::worker(std::_List_iterator<std::thread>) /usr/bin/clickhouse
2019.12.22 19:54:38.984386 [ 189 ] {} <Fatal> BaseDaemon: 16. 0x55986b884e00 ? /usr/bin/clickhouse
2019.12.22 19:54:38.984403 [ 189 ] {} <Fatal> BaseDaemon: 17. 0x7f39feed86db start_thread /lib/x86_64-linux-gnu/libpthread-2.27.so
2019.12.22 19:54:38.984419 [ 189 ] {} <Fatal> BaseDaemon: 18. 0x7f39fe7f588f clone /lib/x86_64-linux-gnu/libc-2.27.so
```

**Comment by alexey-milovidov** [comment]
The same as #9270

**Comment by filimonov** [comment]
Fix  #10071
